atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -90,24 +90,63 @@ public static SerializableTransactionManager create(
             Schema schema,
             Environment env,
             boolean allowHiddenTableAccess) {
-        return create(config, ImmutableSet.of(schema), env, allowHiddenTableAccess);
+        return create(config, Optional.absent(), ImmutableSet.of(schema), env, allowHiddenTableAccess);
+    }
+
+    /**
+     * Create a {@link SerializableTransactionManager} with provided configuration,
+     * {@link SSLSocketFactory}, {@link Schema}, and an environment in which to
+     * register HTTP server endpoints.
+     *
+     * This method is deprecated in favor of specifying the ssl configuration for
+     * client-side leader connections via {@link LeaderConfig}'s sslConfiguration
+     * and not through the sslSocketFactory argument.
```

Comment:
What's random about it?  As an operator of an atlas consuming service I want to know explicitly what the various things I'm setting are that are atlas related and not have to think about the implementation of the consuming service itself.  This is precisely why I even had to do that internal write-up re: leader endpoints, because it wasn't clear when I was testing an internal product how any of that SSL actually getting setup b/c I didn't have knowledge about the underlying implementation of said internal product, and I have a much deeper knowledge of the atlas codebase than your average operator or developer.